Observer/RecommendSettings.php

```diff
@@ -161,38 +128,66 @@ protected function validateTrendingItems(string $changedPath): void
      * @throws LocalizedException
      */
     protected function validateLookingSimilar(string $changedPath): void
+    {
+        $this->validateRecommendation($changedPath, 'getLookingSimilarRecommendation', 'Looking Similar');
+    }
+
+    /**
+     * @param string $changedPath - config path to be reverted if validation failed
+     * @param string $recommendationMethod - name of method to call to retrieve method from RecommendManagementInterface
+     * @param string $modelName - user friendly name to refer to model in error messaging
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateRecommendation(string $changedPath, string $recommendationMethod, string $modelName): void
```

Comment:
Ok, that's what you meant by parameterization. It looks much better for sure. Thanks for spending time and put it together.